Source/BoundaryConditions/WarpX_PML_kernels.H

```diff
@@ -16,113 +16,401 @@ using namespace amrex;
 
 AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
 void warpx_damp_pml_ex (int i, int j, int k, Array4<Real> const& Ex,
+                        amrex::GpuArray<int, AMREX_SPACEDIM> const& Ex_stag,
```

Comment:
Based on [this paragraph](https://amrex-codes.github.io/amrex/docs_html/GPU.html#box-intvect-and-indextype) of the AMReX documentation about Box, IntVect and IndexType, I actually think that I could simply use `amrex::IntVect&` instead of `amrex::GpuArray<int, AMREX_SPACEDIM>&` as argument passed to this kind of kernel functions.

@WeiqunZhang @atmyers Could you confirm this? Thank you.

This would simplify a lot the changes I made in the other file above, Source/BoundaryConditions/WarpXEvolvePML.cpp.

Reply:
Yes, you can do that. `IntVect` is fully functional in device code.